app/main.py

```diff
@@ -1,2 +1,75 @@
+from typing import Hashable, Any
+
+CAPACITY = 8
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash_key = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = CAPACITY) -> None:
+        self.capacity = capacity
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+        self.size = 0
+
+    def calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        return index
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size > self.load_factor * self.capacity:
+            self.resize()
+
+        index = self.calculate_index(key)
+        if self.hash_table[index] is None:
+            self.hash_table[index] = Node(key, value)
+            self.size += 1
+        else:
+            if self.hash_table[index].key == key:
+                self.hash_table[index] = Node(key, value)
+            else:
+                while (
+                    self.hash_table[index] is not None
+                    and self.hash_table[index].key != key
+                ):
+                    index = (index + 1) % self.capacity
```

Comment:
u are repeating this part of code twice in setitem\getitem methods, maybe it is better to move this part to a calc index method ?